odoo/orm/query_builder.py

```diff
@@ -0,0 +1,24 @@
+from odoo.tools.sql import SQL, SQLable
+from odoo.tools.query import Query, IdentifierBuilder
+from .domains import Domain
+
+
+class QueryBuilder(SQLable):
+    def __init__(self, model):
+        self.env = model.env
+        self.model = model
+        self.query = Query(model.env, model._table, model._table_sql)
+        self.domain = Domain([])
+        self.select : list[str | IdentifierBuilder | tuple[str | IdentifierBuilder, str]] = [self.id]
```

Comment:
You could have `model` and `select` on the Query object. Domain just translates into a where clause.
Because Query is already a kind of query builder, this would not be needed.

Reply:
The idea is to be able to still modify the domain (i.e. use the `&` and `|` operators), which is not possible from the Query object